File: src/main/java/com/example/FleetSystem/model/ProductField.java
package com.example.CargoTracking.model;


import com.example.CargoTracking.dto.Type;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.List;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
@Table(name = "product_field")
public class ProductField {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String status;
    private LocalDate createdAt;


    @Enumerated(EnumType.STRING)
    private Type type;

    @OneToMany(mappedBy = "productField", cascade = CascadeType.ALL)
    private List<ProductFieldValues> productFieldValuesList;
}

File: frontend/src/app/modal/ProductField.ts
export interface ProductField {
  createdAt: string | null | undefined;
  id: number | null | undefined;
  name: string | null | undefined;
  productFieldValuesList: ProductFieldValue[];
  status: string | null | undefined;
  type: string | null | undefined;
}

export interface ProductFieldValue {
  id: number | null | undefined;
  name: string | null | undefined;
  status: string | null | undefined;
}
